src/sentry/migrations/0422_backfill_denormalize_notification_actor.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 2.2.28 on 2023-04-17 16:12
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+ACTOR_TEAM_TYPE = 0
+ACTOR_USER_TYPE = 1
+
+
+def backfill_denormalize_notification(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    for ns in RangeQuerySetWrapperWithProgressBar(
+        NotificationSetting.objects.select_related("target").all()
+    ):
+        target = ns.target
+        if target:
+            if target.type == ACTOR_TEAM_TYPE and target.team_id and not ns.team_id:
+                ns.update(team_id=target.team_id)
+            elif target.type == ACTOR_USER_TYPE and target.user_id and not ns.user_id:
+                ns.update(user_id=target.user_id)
```

Comment:
You can't use `.update` in migrations, this exists only in the real models, not the fake django migration versions of them.

Best to use `.save(update_fields=["team_id"])`